lisa/tools/cargo.py

```diff
@@ -56,15 +73,42 @@ def _install(self) -> bool:
                 target=f"{home_dir}/.cargo/bin/cargo",
                 link="/usr/local/bin/cargo",
             )
+            ln.create_link(
+                is_symbolic=True,
+                target=f"{home_dir}/.cargo/bin/rustup",
+                link="/usr/local/bin/rustup",
+            )
         else:
-            raise UnsupportedDistroException(os)
-        return self._check_exists()
+            raise UnsupportedDistroException(node_os)
+        is_installed = self._check_exists()
+
+        # Point cargo from stable toolchain of rust after installation
+        if is_installed:
+            self.node.execute("rustup default stable", shell=True)
+            self.toolchain = self.__get_rust_toolchain()
+            self._log.debug(f"Rust toolchain: {self.toolchain}")
+            self.cargo_command = "".join(
+                [
+                    f"{home_dir}/.rustup/toolchains/",
+                    f"{self.toolchain}/bin/cargo",
+                ]
+            )
+
+        self.node.tools[Rm].remove_file(
+            "/usr/local/bin/cargo",
+            sudo=True,
+        )
+        self.node.tools[Rm].remove_file(
+            "/usr/local/bin/rustup",
+            sudo=True,
+        )
```

Comment:
why remove them?

Reply:
Once it is installed, we will point to PATH of toolchain so we dont need them. This was created while installing so we can pass **_check_exist()** and we can perform **rustup default stable** to set the rust default toolchain.

Reply:
Yes @anirudhrb it is needed as _check_exist() wont take environment variable like PATH as argument and **.cargo** and **.rustup** is getting install under home directory. If we will not create this symlink, _check_exist() will fail. This is what i have faced during the initial development of testcase.